app/topics_handler.go

```diff
@@ -61,87 +61,84 @@ func (th *TopicsHandler) requestTopicWorkers(ctx sdk.Context, topic emissionstyp
 	// This is to avoid requesting inferences for epochs that have already ended
 	workerNonces = synth.FilterNoncesWithinEpochLength(workerNonces, ctx.BlockHeight(), topic.EpochLength)
 
-	maxRetriesToFulfilNoncesWorker := emissionstypes.DefaultParams().MaxRetriesToFulfilNoncesWorker
-	emissionsParams, err := th.emissionsKeeper.GetParams(ctx)
-	if err != nil {
-		Logger(ctx).Warn(fmt.Sprintf("Error getting max retries to fulfil nonces for worker requests (using default), err: %s", err.Error()))
-	} else {
-		maxRetriesToFulfilNoncesWorker = emissionsParams.MaxRetriesToFulfilNoncesWorker
-	}
-	sortedWorkerNonces := synth.SelectTopNWorkerNonces(workerNonces, int(maxRetriesToFulfilNoncesWorker))
-	Logger(ctx).Debug(fmt.Sprintf("Iterating Top N Worker Nonces: %d", len(sortedWorkerNonces)))
-	// iterate over all the worker nonces to find if this is unfulfilled
-	for _, nonce := range sortedWorkerNonces {
-		nonceCopy := nonce
-		Logger(ctx).Debug(fmt.Sprintf("Current Worker block height has been found unfulfilled, requesting inferences %v", nonceCopy))
-		go generateInferencesRequest(ctx, topic.InferenceLogic, topic.InferenceMethod, topic.DefaultArg, topic.Id, topic.AllowNegative, *nonceCopy)
+	if len(workerNonces.Nonces) < 1 {
+		Logger(ctx).Warn("No unfulfilled worker nonces found for topic %d", topic.Id)
+		return
 	}
+	nonceCopy := workerNonces.Nonces[0]
```

Comment:
because we're always adding new nonces to the head of the list, then this should be the latest, no? To be safe / cleaner design, we could abstract this into class methods on the `Nonces` type

Reply:
True, missed that line 
`nonces.Nonces = append([]*types.Nonce{nonce}, nonces.Nonces...)`
so it should work then.

So now the allowed length of this array is governed by the amount of epochLengths while a tx from workerLeader will be accepted. I think this can be lowered from the current value of `MaxUnfulfilledWorkerRequests=100`, to something like 2/3.